src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -221,16 +185,20 @@ private void configChangeHandler(WhatHappened whatHappened, Node node) {
             }
         }
 
-        if (whatHappened == WhatHappened.initialized || node == null) {
-            updateDeviceGroups(whatHappened, deviceGroupTopics);
-            updateCertificateAuthority();
-        } else if (node.childOf(DEVICE_GROUPS_TOPICS)) {
-            updateDeviceGroups(whatHappened, deviceGroupTopics);
-        } else if (node.childOf(CA_TYPE_KEY) || node.childOf(DEPRECATED_CA_TYPE_KEY)) {
-            if (caConfiguration.getCaTypeList().isEmpty()) {
-                return;
+        try {
+            if (whatHappened == WhatHappened.initialized || node == null) {
+                updateDeviceGroups(whatHappened, deviceGroupTopics);
+                UseCases.get(ConfigureCertificateAuthorityUseCase.class).apply(cdaConfiguration);
+            } else if (node.childOf(DEVICE_GROUPS_TOPICS)) {
+                updateDeviceGroups(whatHappened, deviceGroupTopics);
+            } else if (
+                    (node.childOf(CA_TYPE_KEY) || node.childOf(DEPRECATED_CA_TYPE_KEY))
+                            && cdaConfiguration.isCaTypesProvided()
+            ) {
+                UseCases.get(ConfigureCertificateAuthorityUseCase.class).apply(cdaConfiguration);
             }
-            updateCertificateAuthority();
+        } catch (UseCaseException e) {
+            serviceErrored(e);
```

Comment:
gotta be careful with this. At least add a TODO for an escape route to get back to a healthy state.